utils/src/field_helpers.rs

```diff
@@ -112,7 +112,9 @@ impl<F: Field> FieldHelpers<F> for F {
 
         // Pad with zeros until multiple of 32 if necessary because the BigUint to_bytes_le function gives the smallest possible vector of bytes
         if bytes.len() % 32 != 0 {
-            bytes.resize(32 + bytes.len() / 32, 0);
+            // smallest larger multiple of 32 bytes
+            let goal = 32 + bytes.len() / 32;
```

Comment:
this is not going to be a multiple of 32 :D

Reply:
btw why does it have to be a multiple of 32?

Reply:
Oh you're absolutely right! I forgot to multiply by 32 later 😧 I meant:

```
let goal = 32 * (1 + bytes.len() / 32 as usize );
````

The reason for the 32 bytes is to have a uniform format for foreign field elements of the 256-bit field when transforming from and to BigUint. Basically when the elements need less than those 256 bits because the serialization gives the smallest length in bytes possible. If they didn't all have the same format at some point, then the conversions do not lead to the same values.

I will make a quick PR just to fix the 32-multiple issue, and think about other alternative which do not have to pad.

Reply:
Solved in the new branch -> [PR](https://github.com/o1-labs/proof-systems/pull/711)